ddtrace/contrib/_trace_utils_llm.py

```diff
@@ -14,35 +14,89 @@
 from ddtrace.internal.hostname import get_hostname
 from ddtrace.internal.llmobs import LLMObsWriter
 from ddtrace.internal.log_writer import V2LogWriter
+from ddtrace.internal.utils.formats import asbool
 from ddtrace.sampler import RateSampler
 
 
 class BaseLLMIntegration:
     _integration_name = "baseLLM"
 
-    def __init__(self, config, stats_url, site, api_key, app_key=None):
+    def __init__(self, config, stats_url):
         # FIXME: this currently does not consider if the tracer is configured to
         # use a different hostname. eg. tracer.configure(host="new-hostname")
         # Ideally the metrics client should live on the tracer or some other core
         # object that is strongly linked with configuration.
-        self._statsd = get_dogstatsd_client(stats_url, namespace=self._integration_name)
+        self._log_writer = None
+        self._llmobs_writer = None
+        self._statsd = None
         self._config = config
-        self._log_writer = V2LogWriter(
-            site=site,
-            api_key=api_key,
-            interval=float(os.getenv("_DD_%s_LOG_WRITER_INTERVAL" % self._integration_name.upper(), "1.0")),
-            timeout=float(os.getenv("_DD_%s_LOG_WRITER_TIMEOUT" % self._integration_name.upper(), "2.0")),
-        )
-        self._llmobs_writer = LLMObsWriter(
-            site=site,
-            api_key=api_key,
-            app_key=app_key,
-            interval=float(os.getenv("_DD_%s_LLM_WRITER_INTERVAL" % self._integration_name.upper(), "1.0")),
-            timeout=float(os.getenv("_DD_%s_LLM_WRITER_TIMEOUT" % self._integration_name.upper(), "2.0")),
-        )
         self._span_pc_sampler = RateSampler(sample_rate=config.span_prompt_completion_sample_rate)
-        self._log_pc_sampler = RateSampler(sample_rate=config.log_prompt_completion_sample_rate)
-        self._llmobs_pc_sampler = RateSampler(sample_rate=config.llmobs_prompt_completion_sample_rate)
+
+        _dd_api_key = os.getenv("DD_API_KEY", config.get("_api_key"))
+        _dd_app_key = os.getenv("DD_APP_KEY", config.get("_app_key"))
+        _dd_site = os.getenv("DD_SITE", "datadoghq.com")
+
```

Comment:
Should we always define these attributes? This will prevent us from running into attribute errors if the writers are not defined.

```suggestion
        self._log_writer = None
        self._log_pc_sampler = None
        self._llmobs_writer = None
        self._llmobs_pc_sampler
```

Reply:
Good call - we always define the writers, but the sampler instances should never be called unless the corresponding feature is enabled due to the feature flag checks I added before we call these objects.